music_assistant/controllers/media/base.py

```diff
@@ -338,19 +343,22 @@ async def get_provider_mapping(self, item: ItemCls) -> tuple[str, str]:
             for prov_mapping in item.provider_mappings:
                 if not prov_mapping.available:
                     continue
-                if provider := self.mass.get_provider(
+                provider = self.mass.get_provider(
                     prov_mapping.provider_instance
                     if prefer_unique
                     else prov_mapping.provider_domain
-                ):
-                    if prefer_unique and provider.is_streaming_provider:
-                        continue
-                    return (prov_mapping.provider_instance, prov_mapping.item_id)
+                )
+                if not provider:
+                    continue
+                provider = cast("MusicProvider", provider)
+                if prefer_unique and provider.is_streaming_provider:
+                    continue
+                return (prov_mapping.provider_instance, prov_mapping.item_id)
         # last resort: return just the first entry
         for prov_mapping in item.provider_mappings:
             return (prov_mapping.provider_domain, prov_mapping.item_id)
 
-        return (None, None)
+        return ("", "")
```

Comment:
shouldnt we just raise something here instead of returning magic values?

Reply:
Indeed does not seem to be used anywhere. Let's clean it up.

Reply:
Removed